backend/src/auth/oauth2-google-strategy.service.ts

```diff
@@ -0,0 +1,33 @@
+import { StrategyOptionsWithRequest, VerifyCallback } from 'passport-oauth2';
+import { Injectable } from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { Request } from 'express';
+import { ConfigService } from '@nestjs/config';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { PassportStrategy } from '@nestjs/passport';
+
+@Injectable()
+export class OAuth2GoogleStrategy extends PassportStrategy(GoogleStrategy, 'oauth2-google') {
+  constructor(private authService: AuthService, private configService: ConfigService) {
+    super(
+      {
+        clientID: configService.get('oauth2-google.clientId'),
+        clientSecret: configService.get('oauth2-google.clientSecret'),
+        callbackURL: configService.get('oauth2-google.callbackUrl'),
+        passReqToCallback: true,
+        scope: ['profile', 'email'],
+        state: true,
```

Comment:
**question:** how'd you make it work with state?

Reply:
In the end I didn't - tested it now one more time and using `state: true` doesn't work. I changed it back to `state: false`.